src/spandrel/architectures/SwinIR/__init__.py

```diff
@@ -35,30 +34,18 @@ def load(state_dict: StateDict) -> ImageModelDescriptor[SwinIR]:
     upsampler = ""
     resi_connection = "1conv"
     num_feat = 64
-    num_in_ch = in_chans
-    num_out_ch = in_chans
     start_unshuffle = 1
 
-    state_keys = state_dict.keys()
-
-    if "conv_before_upsample.0.weight" in state_keys:
-        if "conv_up1.weight" in state_keys:
+    if "conv_before_upsample.0.weight" in state_dict:
+        if "conv_up1.weight" in state_dict:
             upsampler = "nearest+conv"
         else:
             upsampler = "pixelshuffle"
-    elif "upsample.0.weight" in state_keys:
+    elif "upsample.0.weight" in state_dict:
         upsampler = "pixelshuffledirect"
     else:
         upsampler = ""
 
-    num_feat_pre_layer = state_dict.get("conv_before_upsample.weight", None)
-    num_feat_layer = state_dict.get("conv_before_upsample.0.weight", None)
-    num_feat = (
-        num_feat_layer.shape[1]
-        if num_feat_layer is not None and num_feat_pre_layer is not None
-        else 64
-    )
-
```

Comment:
Why did you get rid of this? Is it incorrect?

Reply:
Assuming that it's because it's hardcoded to 64, I think instead of doing this to all the models we should just add num_feat as a parameter you can change, and then add detection for it.

Reply:
For what reason?

Reply:
It just feels wrong to me to let that be customized on some networks but not on others

Reply:
Eh whatever it's fine